### ISSUE TITLE
don't pin requests to a specific broken version

### ISSUE THREAD

**Issue by johnrfrank** [RaRe-Technologies/smart_open#51]
Was there a reason to pin requests==2.8.1 instead of the much friendlier "<=2.8.1"?  

That version of requests pulls in a broken version of urllib3 that cannot do SSL properly, so we have had to pin "requests<=2.7.0".  Since you pinned to that specific version of requests, I have to also say:

        'gensim <= 0.12.1', # so it doesn't pull in smart-open newest that pins requests to 2.8.1
        'smart-open < 1.3.1', # yuck.
        'requests', # <= 2.7.0', # pinned until this is released: https://github.com/shazow/urllib3/issues/717

If you used "requests<=2.8.1" then everything would work fine... unless you have some need of 2.8.1 specifically in smart-open?

**Comment by tmylk** [comment]
That's a httpretty requirement 2.8.1 so we need it to run tests.

**Comment by johnrfrank** [comment]
I looked in HTTPretty to figure out if they can fix it there, and [they have a requirements.txt file](https://github.com/gabrielfalcao/HTTPretty/blob/master/requirements.txt) that says:

`HTTPretty doesn't have any requirements per se so far. yay!`

so.... where is requests==2.8.1 required?  :-)

**Comment by tmylk** [comment]
See development.txt file 

Would really appreciate if you could make httpretty work on Python 2 and 3 with later request!

**Comment by piskvorky** [comment]
@tmylk not sure I understand. Why are we pulling in httpretty's `development.txt` file? Or even its `requirements.txt` file? How is it smart_open's business what libraries httpretty needs or doesn't need?

**Comment by tmylk** [comment]
An error is generated by httpretty when it is used with requests later version.
I will post it this weekend.

**Comment by tmylk** [comment]
@piskvorky 
This is resolved by fixing the requests version.
This was not an issue in previous version of httpretty so I assume httpretty is the culprit.
Httpretty is developed with requests 2.8.1 version fixed.

```
======================================================================
ERROR: test_write (smart_open.tests.test_smart_open.WebHdfsWriteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<string>", line 3, in wrapper
  File "/home/travis/build/piskvorky/smart_open/smart_open/tests/test_smart_open.py", line 463, in test_write
    smart_open_object.close()
  File "/home/travis/build/piskvorky/smart_open/smart_open/smart_open_lib.py", line 560, in close
    self.upload(buff)
  File "/home/travis/build/piskvorky/smart_open/smart_open/smart_open_lib.py", line 523, in upload
    response = requests.post(uri, data=data, headers={'content-type': 'application/octet-stream'})
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/api.py", line 107, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/api.py", line 53, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/sessions.py", line 454, in request
    prep = self.prepare_request(req)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/sessions.py", line 388, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/models.py", line 296, in prepare
    self.prepare_body(data, files, json)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/models.py", line 447, in prepare_body
    body = self._encode_params(data)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/models.py", line 84, in _encode_params
    return to_native_string(data)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/requests/utils.py", line 700, in to_native_string
    out = string.decode(encoding)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 0: ordinal not in range(128)
```

**Comment by tmylk** [comment]
Apologies, I have incorrectly assumed you were asking for a later version of requests, but now I realise you wanted to use an older one.

This is relaxed and merged in in #57. 

If you could report back if it fixes the issue for you that would be greatly appreciated.

**Comment by tmylk** [comment]
Requests are no longer pinned. 
Closing as abandoned.
